src/containers/layouts/ecommerce/productDetails/byu.modal.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  View,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ListRenderItemInfo,
+  Dimensions,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  List,
+  Button,
+} from '@kitten/ui';
+import { Text } from '@src/components/common';
+import { Product as ProductModel } from '@src/core/model';
+import { BuyActionModel } from './productDetails.container';
+
+const { width, height } = Dimensions.get('window');
+
+interface ComponentProps {
+  product: ProductModel;
+  shoppingItemsCount: number;
+  actionItems: BuyActionModel[];
+  onCancel: () => void;
+}
+
+export type ByuModalProps = ThemedComponentProps & ComponentProps;
+
+class ByuModalComponent extends React.Component<ByuModalProps> {
+
+  private onCancel = (): void => {
+    this.props.onCancel();
+  };
+
+  private renderAction = (info: ListRenderItemInfo<BuyActionModel>): React.ReactElement<TouchableOpacityProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <TouchableOpacity
+        style={themedStyle.actionContainer}
+        activeOpacity={0.95}
+        onPress={info.item.action}>
+        <Text style={themedStyle.actionLabel}>{info.item.title}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  public render(): React.ReactNode {
+    const {
+      themedStyle,
+      shoppingItemsCount,
+      product,
+      actionItems,
+    } = this.props;
+
+    return (
+      [
+        <View style={themedStyle.container} key={0}>
+          <View style={themedStyle.headerContainer}>
+            <Text style={themedStyle.nameLabel}>{product.name}</Text>
+            <Text style={themedStyle.countLabel}>
+              {`You have ${shoppingItemsCount} in your Shopping Cart`}
+            </Text>
+          </View>
+          <List
+            style={themedStyle.actionsList}
+            data={actionItems}
+            renderItem={this.renderAction}
+          />
+          <Button
+            appearance='ghost'
+            size='large'
+            onPress={this.onCancel}>
+            Cancel
+          </Button>
+        </View>,
+        <View
+          style={themedStyle.wrapper}
+          pointerEvents='box-none'
+          key={1}
+        />,
+      ]
+    );
+  }
+}
+
+export const ByuModal = withStyles(ByuModalComponent, (theme: ThemeType) => ({
+  wrapper: {
+    width: width,
+    height: height,
+    backgroundColor: theme['color-black'],
+    opacity: 0.4,
+    position: 'absolute',
+  },
+  container: {
+    zIndex: 1,
+    backgroundColor: theme['color-white'],
+    justifyContent: 'space-between',
+    paddingTop: 24,
+    paddingBottom: 16,
+    width: 343,
+    height: 276,
+    borderRadius: 12,
+    top: (height - 276) / 2,
```

Comment:
this looks crazy, is this a framework issue?

Reply:
This is not an issue, just one of the options of placing 'absolute' View 'screen-centered'. We allow the user to do this for flexibility.